### ISSUE TITLE
terratorch CLI help printout incorrect?

### ISSUE THREAD

**Issue by Foxigod** [IBM/terratorch#108]
**Describe the issue**
The help printout from terratorch CLI seems to be inherited from the PyTorch Lightning docs but the functionality has been modified in terratorch to be inconsistent with said docs. 
Running `terratorch fit -h` produces:
```
[...]
  --trainer.enable_checkpointing {true,false,null}
                        If ``True``, enable checkpointing. It will configure a default ModelCheckpoint callback if there is no user-defined ModelCheckpoint in
                        :paramref:`~lightning.pytorch.trainer.trainer.Trainer.callbacks`. Default: ``True``. (type: Optional[bool], default: null)
[...]
```
which specifically states that a _default_ ModelCheckpoint callback will be created if a ModelCheckpoint callback doesn't already exist in the trainer callbacks. In my experience with terratorch this doesn't actually seem to be a "default" ModelCheckpoint callback, rather it seems to monitor a specific quantity and thus doesn't save a new checkpoint when this monitored quantity doesn't improve. Looking at the [PyTorch Lightning docs](https://github.com/Lightning-AI/pytorch-lightning/blob/1551a16b94f5234a4a78801098f64d0732ef5cb5/src/lightning/pytorch/trainer/connectors/callback_connector.py#L93) it seems to actually instantiate a default instance of ModelCheckpoint. However, since this is not my experience with terratorch, I have to look into the terratorch code where I find this:
![Screenshot 2024-08-10 at 12 45 03](https://github.com/user-attachments/assets/73b6ad89-5198-4ade-b5c9-8b935b740935) Although I don't fully know how all these things interact with one another, this screenshot seems to show a configuration that is consistent with my experience with terratorch, the filename as `{epoch}` and the monitor quantity being set to `val/loss`. The filename and monitor parameters also seem to be consistent with the StateDict checkpoint that gets saved.

So from my point of view it seems that terratorch "blindly" inherits the documentation from Pytorch Lightning but then modifies the functionality to be inconsistent with said documentation.

**To Reproduce (optional, but appreciated)**
1. Run terratorch from CLI with a config file with the `trainer.enable_checkpointing` option set to True and don't include any other ModelCheckpoint callback in the trainer callbacks. (This even seems to be the case for examples like [sen1floods11](https://github.com/IBM/terratorch/blob/main/examples/confs/sen1floods11_vit.yaml))
2. Notice how the saved checkpoints are inconsistent with what was expected.

**Deployment information (optional)**
I installed terratorch by cloning the repo and running `pip3 install -e <cloned_directory>`.
If I run `git rev-parse HEAD` I get the following output:
```
3b763912d54594f5e89092ca5999178b399c2a10
```
which is [this commit](https://github.com/IBM/terratorch/commit/3b763912d54594f5e89092ca5999178b399c2a10).

**Notes**
The issue I am raising here is not just of this specific part of the help printout from terratorch, but of the "integrity" of said printout. There might exist similar issues with other parts that I haven't encountered.

**Comment by Joao-L-S-Almeida** [comment]
Considered as solved.
